includes/class-webfinger.php

```diff
@@ -302,11 +303,24 @@ public static function generate_cache_key( $uri ) {
 	 * Infer a shortname from the Actor ID or URL. Used only for fallbacks,
 	 * we will try to use what's supplied.
 	 *
-	 * @param string $uri The URI.
+	 * @param Actor|string $actor_or_uri The Actor or URI.
 	 *
 	 * @return string Hopefully the name of the Follower.
 	 */
-	public static function guess( $uri ) {
-		return extract_name_from_uri( $uri ) . '@' . \wp_parse_url( $uri, PHP_URL_HOST );
+	public static function guess( $actor_or_uri ) {
+		if ( ! $actor_or_uri instanceof Actor ) {
+			$actor = Actors::fetch_remote_by_uri( $actor_or_uri );
```

Comment:
Making an external request here makes the function much heavier. Are you worried about the function getting in a way where it doesn't get actors passed and inadvertently causes lots of external requests, if used in a loop?

Is the path problem not solved by the change in `extract_name_from_uri()`?